crates/air_r_formatter/src/r/auxiliary/unary_expression.rs

```diff
@@ -11,17 +14,37 @@ impl FormatNodeRule<RUnaryExpression> for FormatRUnaryExpression {
         let operator = operator?;
         let argument = argument?;
 
-        write!(f, [operator.format()])?;
+        let needs_space = is_complex_anonymous_function(&operator, &argument);
 
-        if f.comments().has_comments(argument.syntax())
-            && !f.comments().is_suppressed(argument.syntax())
-        {
-            // We never break between the operator and its argument except if a
-            // newline separates them
-            write!(f, [soft_block_indent(&argument.format())])
-        } else {
-            // A unary operator always sticks to its argument
-            write!(f, [argument.format()])
-        }
+        let argument = format_with(|f| {
+            if f.comments().has_comments(argument.syntax())
+                && !f.comments().is_suppressed(argument.syntax())
+            {
+                // We never line break between the operator and its argument except if an
+                // existing comment separates them
+                write!(f, [soft_block_indent(&argument.format())])
+            } else {
+                write!(f, [argument.format()])
+            }
+        });
+
+        write!(f, [operator.format(), maybe_space(needs_space), argument])
     }
 }
+
+/// Is this an anonymous function like `~ .x + 1`?
+///
+/// Simple unary formulas like `~foo` don't have a space between the `~` and the
+/// argument (often used in tribble calls), but "complex" unary formulas do.
+///
+/// We define "complex" as anything except a simple identifier, so even `~ "foo"` results
+/// in a space.
+fn is_complex_anonymous_function(operator: &RSyntaxToken, argument: &AnyRExpression) -> bool {
+    // Must be a `~`
+    if !matches!(operator.kind(), RSyntaxKind::TILDE) {
+        return false;
+    }
+
+    // "Complex" is anything except identifiers
+    !matches!(argument, AnyRExpression::RIdentifier(_))
```

Comment:
I would expect `~NA` and `~NULL` to be treated like `~foo`?
And probably other literals like `~10` too?